Sources/PointFree/Mailgun/Mailgun.swift

```diff
@@ -0,0 +1,96 @@
+import Either
+import Foundation
+import HttpPipeline
+import Optics
+import Prelude
+
+enum Tracking: String {
+  case no
+  case yes
+}
+
+enum TrackingClicks: String {
+  case yes
+  case no
+  case htmlOnly = "htmlonly"
+}
+
+enum TrackingOpens: String {
+  case yes
+  case no
+  case htmlOnly = "htmlonly"
+}
+
+public struct Email {
+  var from: String
+  var to: [String]
+  var cc: [String]? = nil
+  var bcc: [String]? = nil
+  var subject: String
+  var text: String?
+  var html: String?
+  var testMode: Bool? = nil
+  var tracking: Tracking? = nil
+  var trackingClicks: TrackingClicks? = nil
+  var trackingOpens: TrackingOpens? = nil
+  var domain: String
+}
+
+public struct SendEmailResponse: Decodable {
+  let id: String
+  let message: String
+}
+
+func mailgunSend(email: Email) -> EitherIO<Prelude.Unit, SendEmailResponse> {
+
+  let params = [
+    "from": email.from,
+    "to": email.to.joined(separator: ","),
+    "cc": email.cc?.joined(separator: ","),
+    "bcc": email.bcc?.joined(separator: ","),
+    "subject": email.subject,
+    "text": email.text,
+    "html": email.html,
+    "tracking": email.tracking?.rawValue,
+    "tracking-clicks": email.trackingClicks?.rawValue,
+    "tracking-opens": email.trackingOpens?.rawValue
```

Comment:
Should we make this stuff `Codable`?

Reply:
only complication is that the to/cc/bcc all need to be comma separated in the submitted data, or submitted in a url encoded format like `to=a@a.com&to=b@b.com&...` whereas our form encoder helper would do `to[]=a@a.com&to[]=b@b.com&...` :/ prob means we should make our encoder customizable to allow stuff like that.